Dockerfile

```diff
@@ -72,18 +39,53 @@ COPY config/ config/
 RUN chmod u-w docroot/sites/default/settings.php \
   && chmod u-w docroot/sites/default/services.yml
 
+###########################################################################################
+# Create test image
+###########################################################################################
+
+FROM base AS test
+
+# Remove the memory limit for the CLI only.
+RUN echo 'memory_limit = -1' > /usr/local/etc/php/php-cli.ini
+
+# Install dependencies (with dev)
+RUN composer install \
+  --no-ansi \
+  --dev \
+  --no-interaction \
+  --prefer-dist
+
 # Change ownership of files
 RUN chown -R www-data:www-data ./
 RUN chown -R www-data:www-data /var/www
 
 USER www-data
 
-FROM build as local
+FROM test as local
 USER root
 RUN pecl install xdebug-2.9.8 \
   && docker-php-ext-enable xdebug
 USER www-data
 
+###########################################################################################
+# Create build
+###########################################################################################
+FROM base as build
+
+# Install dependencies
+RUN composer install \
+  --no-ansi \
+  --no-dev \
+  --optimize-autoloader \
+  --no-interaction \
+  --prefer-dist
+
+# Change ownership of files
+RUN chown -R www-data:www-data ./
+RUN chown -R www-data:www-data /var/www
+
+USER www-data
+
 # Make build (and not local) the default target.
 # By ensuring this is the last defined target in the file.
 FROM build as production
```

Comment:
As `build` is now the last step in the `Dockerfile`, we could remove these lines, and (maybe) rename L73 to `production` or `optimised` or something.

Reply:
Good idea. I think `optimised` works well.  I've updated this now in the PR.